mempool/v1/mempool.go

```diff
@@ -401,25 +410,22 @@ func (txmp *TxMempool) Update(
 
 	for i, tx := range blockTxs {
 		// The tx hash corresponding to the originating transaction. Set to the
-		// current tx hash initially, but can change in case of a malleated tx.
-		originalKey := tx.Key()
-
-		// Regardless of outcome, remove the transaction from the mempool.
-		if err := txmp.removeTxByKey(originalKey); err != nil {
-			if malleatedTx, isMalleated := types.UnwrapMalleatedTx(tx); isMalleated {
-				copy(originalKey[:], malleatedTx.OriginalTxHash)
-				_ = txmp.removeTxByKey(originalKey)
-			}
+		// current tx.
+		if malleatedTx, isMalleated := types.UnwrapMalleatedTx(tx); isMalleated {
```

Comment:
> The act or process of beating into a plate, sheet, or leaf, as a metal; extension by beating.

lolol I love that word! but yeah good point it no longer makes sense since we're using a single tx throughout.

Reply:
Relevant https://github.com/celestiaorg/celestia-app/pull/1005/files#diff-e1480f76524405adf851f7b9d0ca187c9e8b24e3211801bd4d02636b51aa5141R66 but on second thought I don't think `ShareIndexedTx` works long term b/c we may add additional metadata to the tx.